modin/core/execution/ray/common/engine_wrapper.py

```diff
@@ -114,7 +117,12 @@ def materialize(cls, obj_id):
         object
             Whatever was identified by `obj_id`.
         """
-        return ray.get(obj_id)
+        materialized = ray.get(obj_id)
+        for obj in materialized if isinstance(materialized, list) else [materialized]:
+            if isinstance(obj, DeferredExecutionException):
```

Comment:
Why did you decide to wrap exceptions in this type?

Reply:
To keep the original exception, occurred, when executing the function.

Reply:
Previously, it was also used to for debugging to get more failure details such as method parameters and the deconstructed list. Replaced it with logger, but to get the messages it should be enabled for the Ray workers first.